examples/postgres/pooled-with-rustls/src/main.rs

```diff
@@ -0,0 +1,62 @@
+use diesel::{ConnectionError, ConnectionResult};
+use diesel_async::pooled_connection::bb8::Pool;
+use diesel_async::pooled_connection::AsyncDieselConnectionManager;
+use diesel_async::AsyncPgConnection;
+use futures_util::future::BoxFuture;
+use futures_util::FutureExt;
+use std::time::Duration;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let db_url = std::env::var("DATABASE_URL").expect("Env var `DATABASE_URL` not set");
+
+    // First we have to construct a connection manager with our custom `establish_connection`
+    // function
+    let mgr = AsyncDieselConnectionManager::<AsyncPgConnection>::new_with_setup(
+        db_url,
+        establish_connection,
+    );
+    // From that connection we can then create a pool, here given with some example settings.
+    let pool = Pool::builder()
+        .max_size(10)
+        .min_idle(Some(5))
+        .max_lifetime(Some(Duration::from_secs(60 * 60 * 24)))
+        .idle_timeout(Some(Duration::from_secs(60 * 2)))
+        .build(mgr)
+        .await?;
+
+    // Now we can use our pool to run queries over a TLS-secured connection:
+    let conn = pool.get().await?;
+    let _ = conn;
+
+    Ok(())
+}
+
+fn establish_connection(config: &str) -> BoxFuture<ConnectionResult<AsyncPgConnection>> {
+    let fut = async {
+        // We first set up the way we want rustls to work.
+        let rustls_config = rustls::ClientConfig::builder()
+            .with_safe_defaults()
+            .with_root_certificates(root_certs())
+            .with_no_client_auth();
```

Comment:
I think it might be a good idea to add a comment here that this is the equivalent setup for libpq's `sslmode=verify-full` so that the next person has an easier time to figure out why this not works with their certs, while it works with `libpq`.